src/Symfony/Bridge/PhpUnit/SymfonyTestsListener.php

```diff
@@ -151,13 +154,37 @@ public function startTest(\PHPUnit_Framework_Test $test)
             if (in_array('dns-sensitive', $groups, true)) {
                 DnsMock::register(get_class($test));
             }
+
+            $annotations = \PHPUnit_Util_Test::parseTestMethodAnnotations(get_class($test), $test->getName(false));
+
+            if (isset($annotations['class']['expectedDeprecation'])) {
+                $test->getTestResultObject()->addError($test, new \PHPUnit_Framework_AssertionFailedError('`@expectedDeprecation` annotations are not allowed at the class level.'), 0);
+            }
+            if (isset($annotations['method']['expectedDeprecation'])) {
+                if (!in_array('legacy', $groups, true)) {
+                    $test->getTestResultObject()->addError($test, new \PHPUnit_Framework_AssertionFailedError('Only tests with the `@group legacy` annotation can have `@expectedDeprecation`.'), 0);
+                }
+                $this->expectedDeprecations = $annotations['method']['expectedDeprecation'];
+                $this->previousErrorHandler = set_error_handler(array($this, 'handleError'));
+            }
         }
     }
 
     public function endTest(\PHPUnit_Framework_Test $test, $time)
     {
+        if ($this->expectedDeprecations) {
+            restore_error_handler();
+            try {
+                $prefix = "@expectedDeprecation:\n  ";
+                $test->assertStringMatchesFormat($prefix.implode("\n  ", $this->expectedDeprecations), $prefix.implode("\n  ", $this->gatheredDeprecations));
```

Comment:
This now differs in one detail from the current implementation: Previously, the order in which you declared multiple expected deprecations did not matter. Now you need to know in which order they will be triggered.